### ISSUE TITLE
[Proposal] QTabs previous/next methods; Animated tab panes

### ISSUE THREAD

**Issue by brindosch** [quasarframework/quasar#1831]
QTabs currently supports just a *switch by name* navigation through `selectTab(NameOfTab)`. This doesn't work well with touch-swipe where you just want to switch to a specific direction
This is a code proposal to implement a left/right method without further quirks to QTabs. 

```javascript
    selectTabDirection(name = 'right'){
       // switch to the right/left neighbor tab based on current tab
       // 1. Get current tab, may be undefined if not found. In this case we do nothing
       const currTab = this.__getTabByName(this.modtabs)
       if(!currTab)
         return
       // 2. get an array of all possible tabs (filter hidden,disabled tabs and other DOM elements)
       const tabs = this.__getAvailableTabs()
       // 3. Get target tab, may be undefined if not found
       const tab = (name === 'right') ? tabs[tabs.indexOf(currTab)+1] : tabs[tabs.indexOf(currTab)-1]
       // 4. set the new tab
       if(tab)
         this.$refs.tabs.selectTab(tab.name)
    },
    __getAvailableTabs(){
      return this.$refs.tabs.$children.filter(child => child.$vnode.componentOptions.tag === 'q-tab' && !child.disable && !child.hidden)
    },
    __getTabByName (value) {
    	 return this.$refs.tabs.$children.find(child => child.name === value && child.$vnode.componentOptions.tag === 'q-tab')
    }
```
### Usage
Call `selectTabDirection(direction) // 'left' or 'right'`
This can be also splitted or merged with selectTab() *Last may be a breaking change, if someone uses these strings as tab name*
### Notes
- If there is no tab found  (end of tabs/ beginning of tabs/ invalid) this method does nothing
- Skips a hidden or disabled qtab and other nodes
- Requires a active tab to work (I think this is intended, no current tab -> no switch)
- Works with v-if on 'q-tab'

Relates also to: https://github.com/quasarframework/quasar/issues/782

**Comment by rstoenescu** [comment]
Hi,

Thanks for taking a short at this. Very detailed description.
It would work great if the order of $children would be same as that in DOM. Unfortunately, last time I checked, when using v-if's on QTab (just an example) it breaks that order. Don't think things have changed, but can you check this example too? Toggle a v-if on a middle QTab.

Thank you for this proposal.

**Comment by brindosch** [comment]
Thank you for your fast response. I've created a fiddle. https://jsfiddle.net/qhr2co6v/54/
Works now also with
- v-show (requires check: `child.$el.style.display !== 'none'` I didn't found another property to get this inline style
- v-if (Sorting is always the same of the tab's. But be aware that the q-tab-pane moves up and down in between. So a reliable check against q-tab is mandatory. Checkout the fiddle console output and how the tab indexes are moving )

I've replaced the class name check against this: `child.$vnode.componentOptions.tag === 'q-tab'`. Is this reliabe also with SSR/PWA? I've no clue about them.

Thank you

**Comment by brindosch** [comment]
Hello @rstoenescu ,
i've refactored it slightly, much smarter now. I removed also the support for v-show. As there is a hidden prop people shouldn't use v-show (Docu note)
Fiddle: https://jsfiddle.net/qhr2co6v/80/

If you are happy with it, will create a PR

**Comment by lilian21** [comment]
+1

**Comment by rstoenescu** [comment]
@brindosch Many apologies. Don't know how I missed the continuation of the conversation. Yes, a PR would be awesome.

**Comment by rstoenescu** [comment]
Will be available in Quasar v0.17.10

**Comment by olivierkes** [comment]
I'm using this right now with quasar v0.17.18, and the `right` or `left` methods do not skip disabled tab, though it does in the original code from @brindosch.

Fiddle: https://jsfiddle.net/6bju1kdo/ 

Should I open a new issue?
